kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/java/org/kie/server/integrationtests/drools/jms/DroolsJmsResponseHandlerIntegrationTest.java

```diff
@@ -0,0 +1,124 @@
+
+        return new ArrayList<Object[]>(Arrays.asList(new Object[][]{
+                {MarshallingFormat.JAXB, jmsConfiguration},
+                {MarshallingFormat.JSON, jmsConfiguration},
+                {MarshallingFormat.XSTREAM, jmsConfiguration}
+        }));
+    }
+
+    @BeforeClass
+    public static void buildAndDeployArtifacts() {
+        KieServerDeployer.buildAndDeployCommonMavenParent();
+        KieServerDeployer.buildAndDeployMavenProject(ClassLoader.class.getResource("/kjars-sources/ruleflow-group").getFile());
+    }
+
+    @Before
+    public void cleanContainers() {
+        disposeAllContainers();
+        createContainer(CONTAINER_ID, releaseId);
+    }
+
+    @Test
+    public void testExecuteSimpleRuleFlowProcessWithAsyncResponseHandler() {
+        Marshaller marshaller = MarshallerFactory.getMarshaller(new HashSet<Class<?>>(extraClasses.values()),
+                configuration.getMarshallingFormat(), client.getClassLoader());
+        ResponseCallback responseCallback = new BlockingResponseCallback(marshaller);
+        ResponseHandler asyncResponseHandler = new AsyncResponseHandler(responseCallback);
+        ruleClient.setResponseHandler(asyncResponseHandler);
+
+        List<Command<?>> commands = new ArrayList<Command<?>>();
+        commands.add(commandsFactory.newSetGlobal(LIST_NAME, new ArrayList<String>(), LIST_OUTPUT_NAME));
+        commands.add(commandsFactory.newStartProcess(PROCESS_ID));
+        commands.add(commandsFactory.newFireAllRules());
+        commands.add(commandsFactory.newGetGlobal(LIST_NAME, LIST_OUTPUT_NAME));
+        BatchExecutionCommand batchExecution = commandsFactory.newBatchExecution(commands, KIE_SESSION);
+
+        assertNoResponse(ruleClient.executeCommandsWithResults(CONTAINER_ID, batchExecution));
+        ExecutionResults result = responseCallback.get(ExecutionResultImpl.class);
+
+        List<String> outcome = (List<String>) result.getValue(LIST_OUTPUT_NAME);
+        assertThat(outcome).isNotNull();
+        assertThat(outcome).containsExactly("Rule from first ruleflow group executed",
+                "Rule from second ruleflow group executed");
+    }
+
+    @Test
+    public void testExecuteSimpleRuleFlowProcessWithFireAndForgetResponseHandler() throws InterruptedException {
+        List<Command<?>> commands = new ArrayList<Command<?>>();
+        commands.add(commandsFactory.newSetGlobal(LIST_NAME, new ArrayList<String>(), LIST_OUTPUT_NAME));
+        commands.add(commandsFactory.newStartProcess(PROCESS_ID));
+        commands.add(commandsFactory.newFireAllRules());
+
+        BatchExecutionCommand batchExecution = commandsFactory.newBatchExecution(commands, KIE_SESSION);
+
+        ruleClient.setResponseHandler(new FireAndForgetResponseHandler());
+        assertNoResponse(ruleClient.executeCommandsWithResults(CONTAINER_ID, batchExecution));
+
+        // wait for commands to be executed
+        Thread.sleep(500);
```

Comment:
We cannot rely on Thread.sleep as it can fail on slower machines or application servers. It could be updated to waiting condition as is in KieServerSynchronization, for example as anonymous class.